### PR TITLE
feature(web): create script to split large files

### PR DESCRIPTION
**Pull request by Si3rr4wow**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/scripts/src/split-file/cursor.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+const IMPORT_START_INDICATOR = Buffer.from('import');
+const TYPE_START_INDICATOR = Buffer.from('/**');
+const BLOCK_START_INDICATORS = ['const', 'export', 'async', 'function'].map((str) =>
+	Buffer.from(str)
+);
+
+const IMPORT_END_INDICATOR = Buffer.from(';');
+const TYPE_END_INDICATOR = Buffer.from('*/');
+
+export class Cursor {
```
Comment (Si3rr4wow) [inline packages/scripts/src/split-file/cursor.js:17]: This guy goes through a file character by character and feeds back what kind of code it's reading (import, type declaration or code block). It fills an internal buffer with the element it's currently reading hen flags when that buffer contains something worth reading out. 

Currently it contains all the rules for distinguishing between different types of block, which makes it look quite gungey. I'd like to extract those but it didn't feel necessary now.

File: packages/scripts/src/split-file/get-code-sections.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ */
+const writeTypesToBlocks = (blocks, typeStrings) =>
+	blocks.map(({ block, typeIndex }) => ({
+		block,
+		type: typeIndex !== null ? typeStrings[typeIndex] : null
+	}));
+/**
+ * @param {import("./cursor.js").Cursor} cursor
+ */
+export const getCodeSections = (cursor) => {
```
Comment (Si3rr4wow) [inline packages/scripts/src/split-file/get-code-sections.js:29]: Puts the various types of blocks into separate arrays for the file writer to use later

File: packages/scripts/src/split-file/get-paths.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import path from 'path';
+
+export const getPaths = () => {
```
Comment (Si3rr4wow) [inline packages/scripts/src/split-file/get-paths.js:3]: Reads the command line args and uses them to create useful paths

File: packages/scripts/src/split-file/write-files.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
+import { existsSync, mkdirSync, writeFileSync } from 'fs';
+import path from 'path';
+import { camelToKebab } from './utils.js';
+
+/**
+ * @param {string} str
+ * @returns {string}
+ */
+const extendPaths = (str) =>
```
Comment (Si3rr4wow) [inline packages/scripts/src/split-file/write-files.js:9]: adds an extra `../` to paths so they point in the right direction

Comment (Si3rr4wow) [reply packages/scripts/src/split-file/write-files.js:9]: Actually I realise now this is dependent on the output dir being one nest deep in the original files dir. I'll clamp down on output dir specification so this always works

File: packages/scripts/src/split-file/write-files.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	)}\n\n${blockWithTypesAndExport}`;
+
+	return dataToWrite;
+};
+
+/**
+ * @param {string} block
+ * @returns {string}
+ */
+const createBlockName = (block) => {
```
Comment (Si3rr4wow) [inline packages/scripts/src/split-file/write-files.js:34]: Plucks the block's name from its declaration

**Comment by fabfab-kc** [comment]
Hey @Si3rr4wow I don't think this is the place for it...
Unless I missed something, it's a local utility that may be useful to refactor code, but why in the `npm` scripts? I would have put it in a private repo / gists to make available to everyone, when needed?

**Comment by Si3rr4wow** [comment]
@fabfab-kc I don't think, in its current state, this splitter has much value outside of this repo. As I mentioned in my comment on the `Cursor` this bakes in rules that are sympathetic to the way code is written for BO ie ES6 import syntax and jsdoc types. As it is this wouldn't even work on front office code. I can make it more configurable (as I said, I'd like to do that) but then this repo will still end up owning the config

**Pull request closed without merging**